### PR TITLE
Update Gradle configuration and GitHub Action workflow

### PR DESCRIPTION
**Pull request by MGaetan89**
## Changes made

- Update the `code_quality.yml` workflow to use the latest actions versions.
- Make Android Lint and Detekt generate SARIF files to be used by the `code_quality.yml` workflow.
- Let the `code_quality.yml` workflow run for any PR, even in draft.
- Remove unused dependencies from each modules.
- Remove unused KAPT Gradle plugin.
- Disable unused features in Android Gradle Plugin (`BuildConfig`, res value and shaders generation).

## Notes

- No project dependencies have been updated in this PR.
- I didn't update the data provider version, as this doesn't need a dedicated release.
- If these changes should be merged, the branch protection rule needs to be updated to use the new validation names (`Android Lint`, `Detekt`, and `Unit Test`).

### CODE REVIEW

**Review by Loic-Dumas — APPROVED** [review]
Seems good to me.
However, I'm not familiar with Github CI. So maybe wait @StaehliJ feedback before merging if it's not urgent.

**Comment by MGaetan89** [comment]
It's not urgent, so I can wait for his review.
I've reuse what we did in [`pillarbox-android`](https://github.com/SRGSSR/pillarbox-android/blob/main/.github/workflows/build.yml) for the CI.

**Pull request merged**
